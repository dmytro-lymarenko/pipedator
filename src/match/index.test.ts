import { match } from './index';

describe('match()', () => {
	it('should validate a string with provided regexp', () => {
		expect(match(/test/).validate('Some test have passed')).toBeNull();
	});

	it('should fail ', () => {
		expect(match(/test/).validate('Some text have be read')).toMatchSnapshot();
	});

	it('should support custom message', () => {
		expect(match(/test/, 'Custom message').validate('Some text have be read')).toMatchSnapshot();
	});
});
