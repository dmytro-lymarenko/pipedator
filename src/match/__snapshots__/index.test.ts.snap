// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match() should fail  1`] = `
Object {
  "children": null,
  "message": "Value should match regexp: /test/",
  "path": "",
}
`;

exports[`match() should support custom message 1`] = `
Object {
  "children": null,
  "message": "Custom message",
  "path": "",
}
`;
